pkg/console/controllers/oauthclientsecret/oauthclientsecret.go

```diff
@@ -0,0 +1,188 @@
+//	- consoles.operator.openshift.io/cluster .status.conditions:
+//		- type=OAuthClientSecretSyncProgressing
+//		- type=OAuthClientSecretSyncDegraded
+type oauthClientSecretController struct {
+	operatorClient v1helpers.OperatorClient
+	secretsClient  corev1clients.SecretsGetter
+
+	authConfigLister      configv1listers.AuthenticationLister
+	consoleOperatorLister operatorv1listers.ConsoleLister
+	configSecretsLister   corev1listers.SecretLister
+	targetNSSecretsLister corev1listers.SecretLister
+}
+
+func NewOAuthClientSecretController(
+	operatorClient v1helpers.OperatorClient,
+	secretsClient corev1clients.SecretsGetter,
+	authnInformer configv1informers.AuthenticationInformer,
+	consoleOperatorInformer operatorv1informers.ConsoleInformer,
+	configSecretsInformer corev1informers.SecretInformer,
+	targetNSsecretsInformer corev1informers.SecretInformer,
+	recorder events.Recorder,
+) factory.Controller {
+	c := &oauthClientSecretController{
+		operatorClient: operatorClient,
+		secretsClient:  secretsClient,
+
+		authConfigLister:      authnInformer.Lister(),
+		consoleOperatorLister: consoleOperatorInformer.Lister(),
+		configSecretsLister:   configSecretsInformer.Lister(),
+		targetNSSecretsLister: targetNSsecretsInformer.Lister(),
+	}
+
+	return factory.New().
+		WithSync(c.sync).
+		WithInformers(
+			authnInformer.Informer(),
+			consoleOperatorInformer.Informer(),
+			configSecretsInformer.Informer(),
+		).
+		WithFilteredEventsInformers(
+			factory.NamesFilter("console-oauth-config"), targetNSsecretsInformer.Informer(),
+		).
+		ToController("OAuthClientSecretController", recorder.WithComponentSuffix("oauthclient-secret-controller"))
+}
+
+func (c *oauthClientSecretController) sync(ctx context.Context, syncCtx factory.SyncContext) error {
+	if shouldSync, err := c.handleManaged(ctx); err != nil {
+		return err
+	} else if !shouldSync {
+		return nil
+	}
+
+	statusHandler := status.NewStatusHandler(c.operatorClient)
+
+	clientSecret, err := c.targetNSSecretsLister.Secrets(api.TargetNamespace).Get("console-oauth-config")
+	if err != nil && !apierrors.IsNotFound(err) {
+		return err
+	}
+
+	authConfig, err := c.authConfigLister.Get("cluster")
+	if err != nil {
+		return fmt.Errorf("failed to retrieve authentication config: %w", err)
+	}
+
+	var secretString string
+	switch authConfig.Spec.Type {
+	case "", configv1.AuthenticationTypeIntegratedOAuth:
+		// in OpenShift controlled world, we generate the client secret ourselves
+		if clientSecret != nil {
+			secretString = secretsub.GetSecretString(clientSecret)
+		}
+		if len(secretString) == 0 {
+			secretString = crypto.Random256BitsString()
+		}
+	case configv1.AuthenticationTypeOIDC:
+		clientConfig := utilsub.GetOIDCClientConfig(authConfig)
+		if clientConfig == nil {
+			// no config, flush the condition and return
+			statusHandler.AddConditions(status.HandleProgressingOrDegraded("OAuthClientSecretSync", "", nil))
```

Comment:
in 4.16, this case needs to make the desired configuration real.  if the desired configuration is "nothing" break it.